src/edge/edgeManager.ts

```diff
@@ -177,6 +177,84 @@ export class EdgeManager {
         }
     }
 
+    public async loadWebView(outputChannel: vscode.OutputChannel) {
+      const panel = vscode.window.createWebviewPanel("IoTEdgeSamples", "Azure IoT Edge Samples", vscode.ViewColumn.One, {
+        enableScripts: true,
+        retainContextWhenHidden: true,
+      });
+      panel.webview.html = this.getWebViewContent("./assets/views/gallery.html");
+      TelemetryClient.sendEvent(Constants.showSampleGalleryEvent);
+      // Handle messages from the webview
+      panel.webview.onDidReceiveMessage(async (message) => {
+        switch (message.command) {
+          case "openSample":
+            if (message.name && message.url) {
+              await vscode.commands.executeCommand("azure-iot-edge.initializeSample", message.name, message.url, message.platform);
+            }
+            break;
+          case "openLink":
+            if (message.url) {
+              TelemetryClient.sendEvent(Constants.openSampleUrlEvent, {Url: message.url});
+              await vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(message.url));
+            }
+            break;
+        }
+      }, undefined, this.context.subscriptions);
+    }
+
+    public getWebViewContent(templatePath: string): string {
+      const resourcePath = path.join(this.context.extensionPath, templatePath);
+      const dirPath = path.dirname(resourcePath);
+      let html = fse.readFileSync(resourcePath, "utf-8");
+
+      html = html.replace(/(<link.*?\shref="|<script.*?\ssrc="|<img.*?\ssrc=")(.+?)"/g, (m, $1, $2) => {
+          return $1 + vscode.Uri.file(path.join(dirPath, $2)).with({ scheme: "vscode-resource" }).toString(true) + "\"";
+      });
+      return html;
+    }
+
+    public async initializeSample(name: string, url: string, platform: string, channel: vscode.OutputChannel) {
+      try {
+        const parentPath = await this.getSolutionParentFolder();
+
+        if (parentPath === undefined) {
+          throw new UserCancelledError();
+        }
+
+        await fse.ensureDir(parentPath);
+        const sampleName: string = await this.inputSampleName(parentPath, name);
+        const samplePath: string = path.join(parentPath, sampleName);
+
+        channel.show();
+        channel.appendLine(`Downloading sample project to ${samplePath}...`);
+        await this.downloadSamplePackage(url, samplePath);
+        channel.appendLine(`Setting default platform to ${platform}...`);
+        await fse.mkdirp(path.join(samplePath, Constants.vscodeFolder));
+        const vscodeSettingPath = path.join(samplePath, Constants.vscodeFolder, Constants.vscodeSettingsFile);

```

Comment:
Minor concern: what if the sample follows the multi-root project structure?

Reply:
Yes, you are right, now the number of samples are little and all controlled by us, so if there are some multi-root sample project, then we can update our code to compatible with them.

Reply:
Why we need to copy settings file? platform is a property in sample? What we should do is call command to set the default platform right?